### PR TITLE
Refine Android media service and notifications

### LINKED ISSUE #2043: [BUG] Unable to install APK with MediaElement on Android below 7.1
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Did you read the "Reporting a bug" section on Contributing file?

- [X] I have read the "Reporting a bug" section on Contributing file: https://github.com/CommunityToolkit/Maui/blob/main/CONTRIBUTING.md#reporting-a-bug

### Current Behavior

When trying to install an APK with CommunityToolkit.Maui.MediaElement package on Android below 7.1, I get the following error during install:
`Failure [INSTALL_PARSE_FAILED_MANIFEST_MALFORMED]`

Exploring logcat, I find the following:

W/PackageManager( 1530): Failed parse during installPackageLI
W/PackageManager( 1530): android.content.pm.PackageParser$PackageParserException: /data/app/vmdl1876558975.tmp/base.apk (at Binary XML file line #17): <service> does not have valid android:name
W/PackageManager( 1530): 	at android.content.pm.PackageParser.parseBaseApk(PackageParser.java:870)
W/PackageManager( 1530): 	at android.content.pm.PackageParser.parseClusterPackage(PackageParser.java:776)
W/PackageManager( 1530): 	at android.content.pm.PackageParser.parsePackage(PackageParser.java:740)
W/PackageManager( 1530): 	at com.android.server.pm.PackageManagerService.installPackageLI(PackageManagerService.java:10446)
W/PackageManager( 1530): 	at com.android.server.pm.PackageManagerService.access$2300(PackageManagerService.java:233)
W/PackageManager( 1530): 	at com.android.server.pm.PackageManagerService$6.run(PackageManagerService.java:8531)
W/PackageManager( 1530): 	at android.os.Handler.handleCallback(Handler.java:739)
W/PackageManager( 1530): 	at android.os.Handler.dispatchMessage(Handler.java:95)
W/PackageManager( 1530): 	at android.os.Looper.loop(Looper.java:135)
W/PackageManager( 1530): 	at android.os.HandlerThread.run(HandlerThread.java:61)
W/PackageManager( 1530): 	at com.android.server.ServiceThread.run(ServiceThread.java:46)

Line #17 of the AndroidManifest file:
`<service android:name="CommunityToolkit.Maui.Media.Services" android:enabled="true" android:exported="false" android:foregroundServiceType="mediaPlayback" />`

I believe this is due to the service name starting with an uppercase character which causes the instalation to fail on older Android versions. Starting with Android 7.1, this issue does not seem to be triggered on install.

### Expected Behavior

Since Maui's minimum Android version is 5.0, the CommunityToolkit should also function properly on all Android versions equal or above this version.

### Steps To Reproduce

1. Create a new Maui application
2. Add CommunityToolkit.Maui.MediaElement package
3. Create an archive
4. Install the archive on an Android version below 7.1 using adb install

### Link to public reproduction project repository

https://github.com/ricardommatias/ReproductionProject

### Environment

```markdown
- .NET MAUI CommunityToolkit: 9.0.2
- OS: Android 5.0
- .NET MAUI: 8.0.70
- MediaElement: 4.0.1
```

### Anything else?

_No response_

### LINKED ISSUE #2048: [BUG] Media Element Button not firing as Expected
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Did you read the "Reporting a bug" section on Contributing file?

- [X] I have read the "Reporting a bug" section on Contributing file: https://github.com/CommunityToolkit/Maui/blob/main/CONTRIBUTING.md#reporting-a-bug

### Current Behavior

When you start MediaElement the buttons on API level 32 and below do not behave as expected.

### Expected Behavior

When you click on Pause/Play it should not just pause but also update the notification icon with current status. It does not. The button does not change if the first pause/play cycle is started from the notification tray. If you start from the player it works fine afterwards.

### Steps To Reproduce

1. Open and run sample from repo.
2. Navigate to MediaElement page.
3. Click on pause in notification tray before doing anything on page.
4. Notice that the icon does not update/change.

### Link to public reproduction project repository

https://github.com/CommunityToolkit/Maui/tree/main/samples

### Environment

```markdown
- .NET MAUI CommunityToolkit: 9.0.2
- OS: All
- .NET MAUI: 8.0.61
- .NET Maui CommunityToolkitMediaElement: 4.0.1
```

### Anything else?

Issue is caused by a combination of threading and string interpolation. Fix incoming.

### PR DESCRIPTION
**Pull request by ne0rrmatrix**
- Bug fix
   
 ### Description of Change ###
Fixes an issue where MediaElement button does not behave as expected.

 ### Linked Issues ###

 - Fixes #2048

 ### PR Checklist ###

 - [x] Has a linked Issue, and the Issue has been `approved`(bug) or `Championed` (feature/proposal)
 - [ ] Has tests (if omitted, state reason in description)
 - [ ] Has samples (if omitted, state reason in description)
 - [x] Rebased on top of `main` at time of PR
 - [x] Changes adhere to [coding standard](https://github.com/CommunityToolkit/Maui/blob/main/CONTRIBUTING.md#contributing-code---best-practices)
 - [ ] Documentation created or updated: https://github.com/MicrosoftDocs/CommunityToolkit/pulls 

 ### Additional information ###

 This commit introduces several enhancements and fixes to the media playback service and notification handling in the Android part of the application. Key changes include:

- Introduced a `NotificationChanged` event for more effective notification management, replacing the `PropertyChanged` event.
- Improved the `OnStartCommand` method to broadcast UI and player updates, enhancing application responsiveness.
- Streamlined `PendingIntent` flags determination for Android S (API level 31) compatibility.
- Enhanced notification initialization and update processes, including dynamic updates through the `NotificationChanged` event and compatibility adjustments for Android versions below Oreo (API level 26).
- Shifted to using predefined constants for intent actions, improving code readability and maintainability.
- Updated the `BroadcastUpdate` method to use `System.Diagnostics.Trace.TraceInformation` for logging.
- Added resource cleanup and event unsubscription in the `Dispose` method to prevent memory leaks.
- Introduced the `NotificationChangedEventArgs` class in the `CommunityToolkit.Maui.Primitives` namespace for structured notification update information.
- Made various adjustments in the `MediaManager` class and related components to support the new notification handling mechanism and ensure Android version compatibility.

### CODE REVIEW

File: src/CommunityToolkit.Maui.MediaElement/Primitives/NotificationChangedEventArgs.cs [hunk 1]
```diff
@@ -0,0 +1,19 @@
+﻿namespace CommunityToolkit.Maui.Primitives;
+/// <summary>
+/// A Class that contains the event arguments for the notification changed event.
+/// </summary>
+sealed class NotificationChangedEventArgs
```
Comment (pictos) [inline src/CommunityToolkit.Maui.MediaElement/Primitives/NotificationChangedEventArgs.cs:5]:
```suggestion
sealed class NotificationChangedEventArgs : EventArgs
```

File: src/CommunityToolkit.Maui.MediaElement/Services/MediaControlsService.android.cs [hunk 2]
```diff
@@ -263,10 +268,12 @@ protected override void Dispose(bool disposing)
 /// </summary>
 sealed class ReceiveUpdates : BroadcastReceiver
 {
-	public string Action = string.Empty;
-
-	public event PropertyChangedEventHandler? PropertyChanged;
-
+	internal readonly struct UpdateNotification
```
Comment (pictos) [inline src/CommunityToolkit.Maui.MediaElement/Services/MediaControlsService.android.cs:271]: why this is a struct?

Comment (ne0rrmatrix) [reply src/CommunityToolkit.Maui.MediaElement/Services/MediaControlsService.android.cs]: Updated. Not sure why I did that.

**Comment by ne0rrmatrix** [comment]
It needs work. Found a few issues.

**Comment by ne0rrmatrix** [comment]
I'm closing this. I am going to start over once media 3 is merged.

**Pull request closed without merging**